modules/decision/decision_worker.py

```diff
@@ -29,13 +31,23 @@ def decision_worker(
     while not controller.is_exit_requested():
         controller.check_pause()
 
-        merged_data: detections_and_odometry.DetectionsAndOdometry = merged_in_queue.queue.get()
+        merged_data: detections_and_odometry.DetectionsAndOdometry = (
+            merged_in_queue.queue.get_nowait()
+        )
         if merged_data is None:
             break
 
+        obstacle_data: obstacle.Obstacle = obstacle_in_queue.queue.get_nowait()
+        if obstacle_data is None:
+            break
+
         result, value = decider.run(merged_data)
         if not result:
             continue
 
+        result, value = decider.run_obstacle_avoidance(obstacle_data)
+        if not result:
+            continue
+
```

Comment:
use the config passed in from main to decide what to run - we're not running both